src/properties.js

```diff
@@ -0,0 +1,145 @@
+class BaseProperty {
+    constructor(object) {
+        for (let key in object) {
+            this[key] = object[key];
```

Comment:
Because you're just using an object parsed from JSON, this isn't an issue, but using `for (variable in obj)` can be somewhat dangerous in other situations.  Basically if your object inherits properties from another object, you'll iterate over all the properties of the parent object.  So if our object was a decedent of another we'd do:

```
for (const key in object) {
    if (object.hasOwnProperty(key)) {
        // Do stuff with key here. 
    }
}
```

Another example of this at the bottom of:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

Thinking about it now, we could do:
```
for (const key of Object.keys(object)) {
    // Do stuff with key
}
```

which is a lot sexier. 

OR..... if you're feeling adventurous, you could try modifying the Object 'class' so that it supports the `of` keyword so that your code looks like:

```
for (const [key, value] of object) {
/// We have both key and value :D
}
```

The implementation of that is left as an exercise to the reader.

All this is to say that, this works perfectly fine and you don't need to change anything if you don't want to.